app/src/main/java/com/duckduckgo/app/browser/BrowserTabFragment.kt

```diff
@@ -1098,7 +1098,9 @@ class BrowserTabFragment : Fragment(), FindListener, CoroutineScope {
 
     private fun requestDownloadConfirmation(pendingDownload: PendingFileDownload, downloadListener: FileDownloadListener) {
         fragmentManager?.let {
-            DownloadConfirmationFragment.instance(pendingDownload, downloadListener).show(it, DOWNLOAD_CONFIRMATION_TAG)
+            if (!it.isStateSaved) {
```

Comment:
I was wondering if we'd be better using `commitAllowingStateLoss()` but I think it makes no difference in the end. 👍

Reply:
Me too! I oscillated between them and then thought that since the dialog won't show in that circumstance because the activity is being killed, this is slightly closer to desired behavior. But I'd be happy with either too.

Reply:
@subsymbolic was this related to any race condition? e.g: calling `requestDownloadConfirmation` when activity was destroyed or being destroyed?